File: Data Structures & Algorithms/two-integer-sum/submission-0.go
func twoSum(nums []int, target int) []int {
	seenMap := map[int]int{}

	for idx, num := range nums {
		x := target - num

		if val, ok := seenMap[x]; ok {
			return []int{val, idx}
		}
		seenMap[num] = idx

	}

	return []int{}
}

File: Data Structures & Algorithms/two-integer-sum/submission-0.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut seen = HashMap::new();

        for (i, cur) in nums.iter().enumerate() {
            let x = target - cur;

            if seen.contains_key(&x) {
                return vec![seen[&x] as i32, i as i32];
            }

            seen.insert(cur, i);
        }

        vec![]
    }
}
